### PR TITLE
Make buffered printf to be a common feature

### PR DESCRIPTION
**Pull request by no1wudi**
@wenyongh  BTW, I can't find the origin definition of `BH_PLATFORM_OPENRTOS`, do you know where it defined ?

### CODE REVIEW

File: core/config.h [hunk 1]
```diff
@@ -276,6 +276,13 @@
 #define BH_ENABLE_GC_VERIFY 0
 #endif
 
+/* Enable buffered printf for builtin libc or not */
+#ifndef BH_BUILTIN_LIBC_BUFFERED_PRINTF
+#define BH_BUILTIN_LIBC_BUFFERED_PRINTF 0
+#define BH_BUILTIN_LIBC_BUFFERED_SIZE 128
+#define BH_BUILTIN_LIBC_BUFFERED_PREFIX
+#endif
```
Comment (wenyongh) [inline core/config.h:284]: There are too many macros in config.h, these macro are only related to libc_builtin_wrapper.c, how about moving them to it?

File: core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c [hunk 2]
```diff
@@ -345,9 +345,10 @@ sprintf_out(int c, struct str_context *ctx)
     return c;
 }
 
-#ifdef BH_PLATFORM_OPENRTOS
-PRIVILEGED_DATA static char print_buf[128] = { 0 };
-PRIVILEGED_DATA static int print_buf_size = 0;
+#if BH_BUILTIN_LIBC_BUFFERED_PRINTF != 0
+BH_BUILTIN_LIBC_BUFFERED_PREFIX static char
+    print_buf[BH_BUILTIN_LIBC_BUFFERED_SIZE] = { 0 };
+BH_BUILTIN_LIBC_BUFFERED_PREFIX static int print_buf_size = 0;
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c:351]: How about:
```C
#ifndef BUILTIN_LIBC_BUFFERED_PRINTF
#define BUILTIN_LIBC_BUFFERED_PRINTF 0
#endif

#if BUILTIN_LIBC_BUFFERED_PRINTF != 0
#ifndef BUILTIN_LIBC_BUFFERED_PRINT_SIZE
#define BUILTIN_LIBC_BUFFERED_PRINT_SIZE 128
#endif
#ifndef BUILTIN_LIBC_BUFFERED_PRINT_PREFIX
#define BUILTIN_LIBC_BUFFERED_PRINT_PREFIX
#endif

BUILTIN_LIBC_BUFFERED_PRINT_PREFIX static char
    print_buf[BUILTIN_LIBC_BUFFERED_PRINT_SIZE] = { 0 };
BUILTIN_LIBC_BUFFERED_PRINT_PREFIX static int print_buf_size = 0;
...
```

File: core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c [hunk 3]
```diff
@@ -884,7 +885,8 @@ __cxa_allocate_exception_wrapper(wasm_exec_env_t exec_env, uint32 thrown_size)
 
 static void
 __cxa_begin_catch_wrapper(wasm_exec_env_t exec_env, void *exception_object)
-{}
+{
+}
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c:889]: Should not change it, or CI code guideline will report error.

**Comment by wenyongh** [comment]
> @wenyongh BTW, I can't find the origin definition of `BH_PLATFORM_OPENRTOS`, do you know where it defined ?

`BH_PLATFORM_OPENRTOS` is used in another project which isn't open sourced, but it should be OK to apply similar changes to it.

File: core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c (hunk 3 above)
Comment (no1wudi) [reply core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c]: Emm, it was changed by clang-format

File: core/config.h (hunk 1 above)
Comment (no1wudi) [reply core/config.h]: OK

File: core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c [hunk 4]
```diff
@@ -345,9 +345,21 @@ sprintf_out(int c, struct str_context *ctx)
...
+#ifndef BUILTIN_LIBC_BUFFERED_PRINT_SIZE
+#define BUILTIN_LIBC_BUFFERED_PRINT_SIZE 128
+#endif
+#ifndef BUILTIN_LIBC_BUFFERED_PRINT_PREFIX
+#define BUILTIN_LIBC_BUFFERED_PRINT_PREFIX
+#endif
+
+BH_BUILTIN_LIBC_BUFFERED_PREFIX static char
+    print_buf[BH_BUILTIN_LIBC_BUFFERED_SIZE] = { 0 };
+BH_BUILTIN_LIBC_BUFFERED_PREFIX static int print_buf_size = 0;
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c:362]: Should remove BUILTIN_LIBC_BUFFERED_PRINT_PREFIX and BUILTIN_LIBC_BUFFERED_PRINT_SIZE

File: core/shared/platform/nuttx/platform_internal.h [hunk 5]
```diff
@@ -77,6 +77,10 @@ typedef sem_t korp_sem;
 #define CONFIG_HAS_ISATTY 0
 #endif
 
+#define BH_BUILTIN_LIBC_BUFFERED_PRINTF 1
+#define BH_BUILTIN_LIBC_BUFFERED_SIZE 128
+#define BH_BUILTIN_LIBC_BUFFERED_PREFIX
```
Comment (wenyongh) [inline core/shared/platform/nuttx/platform_internal.h:82]: Same as above

Comment (no1wudi) [reply core/shared/platform/nuttx/platform_internal.h]: Sure, forget to rename these symbols

**Pull request merged**